src/mainwindow.cpp

```diff
@@ -2123,6 +2123,8 @@ void MainWindow::loadNoteDirectoryList() {
 
         ui->noteSubFolderTreeWidget->buildTreeForParentItem(noteFolderItem);
         noteFolderItem->setExpanded(true);
+        connect(ui->noteTreeWidget, &QTreeWidget::itemExpanded, ui->noteSubFolderTreeWidget, &NoteSubFolderTree::onItemExpanded);
```

Comment:
Since `loadNoteDirectoryList()` is called quite often, aren't those `connect` statements executed multiple times too?

Reply:
Yeah they should probably be moved someplace better. But I'm wondering why in the firstplace is the note list used for the tree, and not the directory view. Then no extra connect calls would be needed.

Reply:
It doesn't even make sense to me that the NoteSubFolderTree is involved at all, the notes are also added to the NoteSubFolderTree now. I suspect that might be a big regression from a pull request from two years ago that moved lots of stuff around.

Reply:
> the notes are also added to the NoteSubFolderTree now

Ok, the note sub-folder tree is disabled when I restart the application. Still strange.

Reply:
> But I'm wondering why in the firstplace is the note list used for the tree, and not the directory view.

Because the directory view lacks far more interaction for making notes work there than the notetree widget.

Reply:
> Because the directory view lacks far more interaction for making notes work there than the notetree widget.
Got it. I guess NoteSubFolderTree is involved because the `buildTreeForParentItem` method was found useful. It confused me at first since it's getting called on an item that's not a part of the NoteSubFolderTree, and I'm surprised it works. But that's fine. Anyway, I moved the connect() calls to a better place in the last commit.

Reply:
I think a lot got mixed up in the "cleanup contribution" two years ago...